src/main/groovy/pl/allegro/tech/build/axion/release/domain/properties/VersionProperties.groovy

```diff
@@ -17,9 +17,11 @@ class VersionProperties {
     final Closure<Version> versionIncrementer
 
     final boolean sanitizeVersion
-    
+
     final boolean useHighestVersion
 
+    final MonorepoProperties monorepoProperties
```

Comment:
i think `MonorepoProperties` deserves to be a top-level `Properties` object and doesn't have to be nested in `VersionProperties`, but since this is internal object, it can be changed at any time :)

Reply:
`VersionService` currently receives in `currentVersion()` and `currentDecoratedVersion()` a `VersionProperties` object.  To me, this seems quite clean, in that `VersionService` only receives properties that are related to itself and its behaviour is not changed by properties outside of its concern (e.g. it doesn't receive a `HooksProperties` object, for instance).  `VersionService` in turn passes these properties into `VersionResolver`.

If we were to move `MonorepoProperties` out of `VersionProperties` and into `Properties` then `VersionService` would need to have passed to it either the full `Properties` object or both  `MonorepoProperties` and `VersionProperties`.  I'm happy to make the change if you'd like, but it seems to me to result in somewhat more unwieldy service classes - what do you think?

Reply:
I feel convinced. Let's keep it as it is :)

Reply:
Thanks :-)